simulation/simulation.py

```diff
@@ -0,0 +1,295 @@
+        w.pack()
+        try:
+            while True:
+                w.delete("all")
+                now = time.time()
+                for lot in self.lots:
+                    # TODO no need to redraw these every time
+                    lotlat, lotlong = lot.get_location()
+                    w.create_rectangle(lotlat, lotlong, lotlat + 20, lotlong + 20, width=0, fill="green")
+
+                for car in self.cars:
+                    if car.drawing:
+                        dotlat, dotlong = car.get_position(now)
+                        w.create_oval(dotlat, dotlong, dotlat + 5, dotlong + 5, width=0, fill='blue')
+                root.update()
+                yield from asyncio.sleep(interval)
+        except tk.TclError as e:
+            if "application has been destroyed" not in e.args[0]:
+                raise
+
+
+class Waypoint:
+    def __init__(self, timestamp, lat, long):
+        self.time = timestamp
+        self.lat = lat
+        self.long = long
+
+
+class Car:
+    def __init__(self, lat, long):
+        self.lat = lat
+        self.long = long
+        self.destX = 0
+        self.destY = 0
+        self.aDestX = 0
+        self.aDestY = 0
+        self.drawing = False
+        self.speed = 60
+        self.waypoints = []
+        self.waypoints.append(Waypoint(time.time(), self.lat, self.long))
+
+    def get_location(self):
+        # TODO update to use waypoints
+        return self.lat, self.long
+
+    def distance_to(self, x, y):
+        return math.sqrt((self.lat - x)**2 + (self.long - y)**2)
+
+    def get_position(self, now):
+        if len(self.waypoints) > 1:
+            latdiff = self.waypoints[-1].lat - self.waypoints[-2].lat
+            longdiff = self.waypoints[-1].long - self.waypoints[-2].long
+            timediff = self.waypoints[-1].time - self.waypoints[-2].time
+            progress = (now - self.waypoints[-1].time) / timediff
+            poslat = self.waypoints[-2].lat + (latdiff * progress)
+            poslong = self.waypoints[-2].long + (longdiff * progress)
+            return poslat, poslong
+        else:
+            return self.lat, self.long
+
+    def set_initial_destination(self, x, y):
+        self.destX = x
+        self.destY = y
+        # for the moment, assuming no congestion and constant speed, we can calculate the arrival time
+        # however might need to pass it in when things get more complicated
+        newtime = time.time() + (self.distance_to(x, y) / self.speed)
+        self.waypoints.append(Waypoint(newtime, x, y))
+
+    def get_initial_destination(self):
+        return self.destX, self.destY
+
+    def set_allocated_destination(self, x, y):
+        self.aDestX = x
+        self.aDestY = y
+        newtime = time.time() + (self.distance_to(x, y) / self.speed)
+        self.waypoints.append(Waypoint(newtime, x, y))
+
+    def get_allocated_destination(self):
+        return self.aDestX, self.aDestY
+
+
+class ParkingLot:
```

Comment:
There's a lot of duplication here with `parking.shared.rest_models.ParkingLot`

This could just subclass that and add the simulated data, and the allocation methods (or better yet, the `SimManager` should perform the allocation and just update the stored parking lot object).